src/compiler/binder.ts

```diff
@@ -3697,9 +3697,13 @@ namespace ts {
                 break;
 
             case SyntaxKind.NumericLiteral:
-                if ((<NumericLiteral>node).numericLiteralFlags & TokenFlags.BinaryOrOctalSpecifier) {
+                const flags = (<NumericLiteral>node).numericLiteralFlags;
+                if (flags & TokenFlags.BinaryOrOctalSpecifier) {
                     transformFlags |= TransformFlags.AssertES2015;
                 }
+                if (flags & TokenFlags.BigInt) {
+                    transformFlags |= TransformFlags.AssertESNext;
+                }
                 break;

```

Comment:
Do we need to handle `checkStrictModeNumericLiteral` for bigInt?

Reply:
Don't think there are any differences between `bigint` literals in strict and non-strict mode.